src/main/java/net/sf/rails/game/financial/ShareSellingRound.java

```diff
@@ -149,87 +149,101 @@ public void setSellableShares() {
+                    number += company.getPresidentsShare().getShares();
+                    // but limit this to the pool
+                    number = Math.min(number, poolAllowsShares);
+                    log.debug("Dump is possible increased single shares to " + number);
                 }
-            }
-            // TODO The above ignores that a dumped player must be
-            // able to exchange the president's share.
 
-            /*
-             * Check the price. If a cert was sold before this turn, the
-             * original price is still valid
-             */
-            if (sellPrices.containsKey(company)) {
-                price = (sellPrices.get(company)).getPrice();
-            } else {
-                price = company.getMarketPrice();
-            }
+                if (number == 0) {
+                    continue;
+                }
 
-            for (int shareSize = 1; shareSize <= 4; shareSize++) {
-                number = shareCountPerUnit[shareSize];
-                if (number == 0) continue;
-                number =
-                    Math.min(number, maxShareToSell
-                            / (shareSize * company.getShareUnit()));
-                if (number == 0) continue;
+                // Check against the maximum share that can be sold
+                number = Math.min(number, maxShareToSell / shareSize);
 
+                if (number <= 0) {
+                    continue;
+                }
                 // May not sell more than is needed to buy the train
-                while (number > 0 && ((number - 1) * price) > cashToRaise.value())
-                    number--;
-
-                if (number > 0) {
-                    for (int i=1; i<=number; i++) {
-                        sellableShares.add(new SellShares(company, shareSize, i, price));
+                while (number > 0 && ((number - 1) * (price/ company.getShareUnitsForSharePrice())) > cashToRaise.value()) number--;
+                for (int i = 1; i <= number; i++) {
+                    if (checkIfSplitSaleOfPresidentAllowed()) {
+                        // check if selling would dump the company
+                        if (dumpIsPossible && i * shareSize >= dumpThreshold) {
+                            // dumping requires that the total is in the possibleSharesToSell list and that shareSize == 1
+                            // multiple shares have to be sold separately
+                            if (shareSize == 1 && possibleSharesToSell.contains(i * shareSize)) {
+                                sellableShares.add(new SellShares(company, shareSize, i, price, 1));
+                            }
+                        } else {
+                            // ... no dumping: standard sell
+                            sellableShares.add(new SellShares(company, shareSize, i, price, 0));
+                        }
+                    } else if (dumpIsPossible && i * shareSize >= dumpThreshold) {
```

Comment:
Maybe extract out the `i * shareSize` math as it looks to be used in a couple of places?

Reply:
next project :) you are right